src/tests/system/tests/test_authentication.py

```diff
@@ -290,3 +292,71 @@ def test_authentication__user_login_with_overriding_home_directory(
         result = ssh.run("pwd").stdout
         assert result is not None, "Getting path failed!"
         assert result == home_exp, f"Current path {result} is not {home_exp}!"
+
+
+@pytest.mark.topology(KnownTopologyGroup.AnyProvider)
+@pytest.mark.parametrize("method", ["ssh", "su"])
+@pytest.mark.parametrize("sssd_service_user", ("root", "sssd"))
+@pytest.mark.importance("critical")
+@pytest.mark.require(
+    lambda client, sssd_service_user: ((sssd_service_user == "root") or client.features["non-privileged"]),
+    "SSSD was built without support for running under non-root",
+)
+def test_authentication__user_login_with_modified_PAM_stack_provider_is_offline(
+    client: Client, provider: GenericProvider, method: str, sssd_service_user: str
+):
+    """
+    :title: Authenticate with modified PAM when the provider is offline
+    :setup:
+        1. Create user
+        2. Configure SSSD with "cache_credentials = true" and "krb5_store_password_if_offline = true" and
+            "offline_credentials_expiration = 0"
+        3. Modify PAM configuration files /etc/pam.d/system-auth and /etc/pam.d/password-auth
```

Comment:
Hi,

thank you for the test. It is working as expected for me. I think it would be good to add the kind of change of the PAM configuration here as well. So something like e.g. "... so that pam_sss.so is using the 'use_first_pass' option and let another PAM module ask for the password."

bye,
Sumit